### PR TITLE
SYNCOPE-808 Added Nuwan's #27 PR

### PR DESCRIPTION
**Pull request by tmess567**
Please check and inform about any required changes.

### CODE REVIEW

File: ide/netbeans/pom.xml [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+KIND, either express or implied.  See the License for the
+specific language governing permissions and limitations
+under the License.
+-->
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+    <parent>
+        <groupId>org.apache.syncope</groupId>
+        <artifactId>syncope</artifactId>
+        <version>2.0.0-SNAPSHOT</version>
```
Comment (ilgrosso) [inline ide/netbeans/pom.xml]: We're on 2.0.2-SNAPSHOT currently.

File: ide/netbeans/pom.xml [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+    </parent>
+    <artifactId>netbeans-plugin</artifactId>
+    <name>Apache Syncope Netbeans Plugin</name>
+    <packaging>nbm</packaging>
+    <build>
+        <plugins>
+            <plugin>
+                <groupId>org.codehaus.mojo</groupId>
+                <artifactId>nbm-maven-plugin</artifactId>
+                <version>3.13</version>
```
Comment (ilgrosso) [inline ide/netbeans/pom.xml]: Plugin and dependency version should be set in the root pom

File: ide/netbeans/pom.xml [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+                <artifactId>maven-jar-plugin</artifactId>
+                <version>3.0.0</version>
+                <configuration>
+                    <archive>
+                        <manifestFile>${project.build.outputDirectory}/META-INF/MANIFEST.MF</manifestFile>
+                    </archive>
+                </configuration>
+            </plugin>
+            <plugin>
+                <groupId>org.codehaus.mojo</groupId>
```
Comment (ilgrosso) [inline ide/netbeans/pom.xml]: Why skipping IANAL?

File: ide/netbeans/src/main/java/org/apache/syncope/netbeans/plugin/view/ResourceExplorerTopComponent.java [hunk 4]
```diff
@@ -0,0 +1,561 @@
...
+import org.apache.syncope.common.lib.types.ReportTemplateFormat;
+import org.apache.syncope.netbeans.plugin.connector.ResourceConnector;
+import org.apache.syncope.netbeans.plugin.constants.PluginConstants;
+import org.apache.syncope.netbeans.plugin.service.MailTemplateManagerService;
+import org.apache.syncope.netbeans.plugin.service.ReportTemplateManagerService;
+import org.netbeans.api.editor.EditorRegistry;
+import org.netbeans.api.progress.ProgressHandle;
+import org.netbeans.api.progress.ProgressHandleFactory;
+import org.netbeans.api.settings.ConvertAsProperties;
+//import org.netbeans.core.spi.multiview.MultiViewDescription;
```
Comment (ilgrosso) [inline ide/netbeans/src/main/java/org/apache/syncope/netbeans/plugin/view/ResourceExplorerTopComponent.java]: Commented imports?

File: ide/netbeans/src/main/java/org/apache/syncope/netbeans/plugin/view/ResourceExplorerTopComponent.java [hunk 5]
```diff
@@ -0,0 +1,561 @@
...
+    @Override
+    public void componentClosed() {
+        // TODO add custom code on component closing
+    }
+
+    void writeProperties(final java.util.Properties p) {
+        // better to version settings since initial version as advocated at
+        // http://wiki.apidesign.org/wiki/PropertyFiles
+        p.setProperty("version", "1.0");
+        // TODO store your settings
```
Comment (ilgrosso) [inline ide/netbeans/src/main/java/org/apache/syncope/netbeans/plugin/view/ResourceExplorerTopComponent.java]: TODO?

File: ide/netbeans/src/main/java/org/apache/syncope/netbeans/plugin/view/ResourceExplorerTopComponent.java [hunk 6]
```diff
@@ -0,0 +1,561 @@
...
+    }
+
+    void readProperties(final java.util.Properties p) {
+        String version = p.getProperty("version");
+        // TODO read your settings according to their version
```
Comment (ilgrosso) [inline ide/netbeans/src/main/java/org/apache/syncope/netbeans/plugin/view/ResourceExplorerTopComponent.java]: TODO?

File: ide/netbeans/src/main/nbm/manifest.mf [hunk 7]
```diff
@@ -0,0 +1,5 @@
+Bundle-License: http://www.apache.org/licenses/LICENSE-2.0.txt
```
Comment (ilgrosso) [inline ide/netbeans/src/main/nbm/manifest.mf]: lowercase `manifest.mf`?

File: ide/netbeans/src/main/resources/org/apache/syncope/checkstyle.xml [hunk 8]
```diff
@@ -0,0 +1,223 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (ilgrosso) [inline ide/netbeans/src/main/resources/org/apache/syncope/checkstyle.xml]: Why is this file needed at all?

File: ide/netbeans/src/main/resources/org/apache/syncope/netbeans/plugin/bundle.properties [hunk 9]
```diff
@@ -0,0 +1,17 @@
+#Licensed to the Apache Software Foundation (ASF) under one
```
Comment (ilgrosso) [inline ide/netbeans/src/main/resources/org/apache/syncope/netbeans/plugin/bundle.properties]: Why is this file needed at all?

File: ide/netbeans/src/main/resources/org/apache/syncope/netbeans/plugin/service/bundle.properties [hunk 10]
```diff
@@ -0,0 +1,17 @@
+#Licensed to the Apache Software Foundation (ASF) under one
```
Comment (ilgrosso) [inline ide/netbeans/src/main/resources/org/apache/syncope/netbeans/plugin/service/bundle.properties]: Why is this file needed at all?

**Comment by ilgrosso** [comment]
Glad to see you back at work on Syncope @tmess567!

File: ide/netbeans/src/main/java/org/apache/syncope/netbeans/plugin/connector/ResourceConnector.java [hunk 11]
```diff
@@ -0,0 +1,69 @@
...
+import org.apache.syncope.netbeans.plugin.entity.UserProperties;
+import org.apache.syncope.netbeans.plugin.service.MailTemplateManagerService;
+import org.apache.syncope.netbeans.plugin.service.ReportTemplateManagerService;
+
+public final class ResourceConnector {
+
+    private static MailTemplateManagerService MAIL_TTEMPLATE_MANAGER_SERVICE;
+    private static ReportTemplateManagerService REPORT_TEMPLATE_MANAGER_SERVICE;
+
+    private ResourceConnector() {
```
Comment (tmess567) [inline ide/netbeans/src/main/java/org/apache/syncope/netbeans/plugin/connector/ResourceConnector.java]: [ERROR] src/main/java/org/apache/syncope/netbeans/plugin/connector/ResourceConnector.java:[30,1] (design) HideUtilityClassConstructor: Utility classes should not have a public or default constructor.

File: ide/netbeans/pom.xml (hunk 3 above)
Comment (tmess567) [inline ide/netbeans/pom.xml]: [ERROR] Failed to execute goal org.codehaus.mojo:ianal-maven-plugin:1.0-alpha-1:verify-legal-files (default) on project syncope-ide-netbeans: Artifact does not contain any legal files: syncope-ide-netbeans-2.1.0-SNAPSHOT.nbm -> [Help 1]

**Comment by tmess567** [comment]
See #39
